Modules/Devices/test/Devices.Application.Tests/Tests/Extensions/IEnumerableExtensionsTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Backbone.Modules.Devices.Application.Extensions;
+using FluentAssertions;
+using Xunit;
+using static System.Double;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Extensions;
+public class IEnumerableExtensionsTests
+{
+    [Fact]
+    public void Null_Parameter_Throws_ArgumentNullException()
+    {
+        // Act
+        var acting = () => ((List<int>)null).Split(10); ;
+
+        // Assert
+        acting.Should().Throw<ArgumentNullException>();
+    }
+
+    [Fact]
+    public void Empty_List_Returns_Empty_List_Of_Lists()
+    {
+        // Arrange
+        var list = new List<int>().AsReadOnly();
+
+        // Act
+        var result = list.Split(10);
+
+        // Assert
+        result.Should().BeEmpty("The argument list was empty.");
+    }
+
+    [Fact]
+    public void Large_List_Returns_Lists_Of_Lists()
```

Comment:
According to the title, if I call `Split` on a list with 1,000,000 items (which is a large list, I suppose?), it always returns multiple lists. But what if I pass 2,000,000 as batch size?

A better name would be "List_bigger_than_batch_size_returns_multiple_lists"